res/controllers/Reloop-Jockey-3-ME-scripts.js

```diff
@@ -0,0 +1,444 @@
+    }
+    else {    // If button up
+        engine.scratchDisable(currentDeck);
+    }
+}
+
+// The wheel that actually controls the scratching
+Jockey3ME.wheelTurn = function (channel, control, value, status, group) {
+  var newValue=(value-64);
+  var currentDeck = parseInt(group.substring(8,9));
+    // See if we're scratching. If not, skip this.
+    if (!engine.isScratching(currentDeck)) {
+      if (newValue > 1 || newValue < -1) {
+        newValue /= 2;  // If Jogwheel Resolution is High
+      }
+      engine.setValue(group, "jog", newValue);
+      return;
+   }
+    engine.scratchTick(currentDeck,newValue);
+}
+
+// Hotcues
+Jockey3ME.hotcue_activate = function (channel, control, value, status, group) {
+  Jockey3ME.hotc = Jockey3ME.hotcue_buttonSelect(control);
+
+  if (!Jockey3ME.hotcueClearVal && engine.getValue(group,"hotcue_"+Jockey3ME.hotc+"_enabled") == 0 && value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",0);
+  } else if (Jockey3ME.hotcueClearVal && engine.getValue(group,"hotcue_"+Jockey3ME.hotc+"_enabled") == 1 && value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_clear",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_clear",0);
+  } else if (value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",0);
+  }
+}
+
+Jockey3ME.hotcue_buttonSelect = function (control) {
+  Jockey3ME.hotc_midino = [11,12,13,14,15,16,17,18];
+  for (var i = 0; i < Jockey3ME.hotc_midino.length; i++) {
+    switch (control) {
+      case Jockey3ME.hotc_midino[i]:
+        return Jockey3ME.hotc_midino[i] - 10;
+        break;
+    }
+  };
+}
+
+Jockey3ME.hotcueClearVal_off = function() {
+  if (Jockey3ME.hotcueClearVal) {
+    Jockey3ME.hotcueClearVal = 0;
+  };
+}
+
+Jockey3ME.hotcue_clear = function (channel, control, value, status, group) {
+   if (control == 0x09 && value == 0x7F) {
+    Jockey3ME.hotcueClearVal_off();
+    Jockey3ME.hotcueClearVal = 1;
+    midi.sendShortMsg(status,control,0x01);
+   } else {
+    Jockey3ME.hotcueClearVal_off();
+    midi.sendShortMsg(status,control,0x00);
+   };
+}
+
+// Effect Sections
+Jockey3ME.effectParam = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(23,24));
+  var EncoderKnopDryWet = 0;
+  var EncoderKnopFX = 0;
+  var newVal = 0;
+  var interval = 0.04;
+  var min = 0;
+  var max = 1;
+  switch (control) {
+    case 29:
+      EncoderKnopDryWet = 1;
+      break;
+    default:
+      EncoderKnopFX = control - 29;
+      break;
+  }
+  if (!EncoderKnopDryWet) {
+    if (value == 0x41) {
+      var curVal = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "parameter" + EncoderKnopFX);
+      newVal = curVal + interval;
+      if (newVal > max) newVal = max;
```

Comment:
Is this logic necessary? Does engine.setParameter not work as expected if it is set to a value outside of 0 through 1?